python/test/regression/test_cast_matmul.py

```diff
@@ -12,12 +12,18 @@
 import triton
 import triton.language as tl
 
+current_target = triton.runtime.driver.active.get_current_target()
 input_dtypes = ["float16", "float32", "float64"]
-if triton.runtime.driver.active.get_current_target().backend == "cuda":
+if current_target.backend == "cuda":
     input_dtypes += ["int8", "float8_e5m2"]
     cc = torch.cuda.get_device_capability(0)
     if cc >= (8, 9):
         input_dtypes += ["float8_e4m3fn"]
+elif current_target.backend == "hip":
```

Comment:
You might want to use `is_hip_mi300` from `_internal_testing.py` here. the fp8 variant is only natively supported there.

Reply:
Actually maybe fine to enable for mi200 too given SW emulation is there. But could use the helper methods in `_Internal_testing.py` to make it a bit nicer.

Reply:
I don't think these cases will run on mi200? Maybe I'm misunderstanding https://rocm.docs.amd.com/en/latest/reference/gpu-arch-specs.html but 940, 941, 942 are all mi300? But I can change to `is_hip_mi300`.

Reply:
Oh sorry you mean emulation for `float8_e5m2` on mi200. gotcha. yea I think it's better not test the SW emulation paths because they're flaky right now.